pandas/core/dtypes/dtypes.py

```diff
@@ -483,67 +484,63 @@ class DatetimeTZDtype(PandasExtensionDtype):
     str = '|M8[ns]'
     num = 101
     base = np.dtype('M8[ns]')
+    na_value = NaT
     _metadata = ('unit', 'tz')
     _match = re.compile(r"(datetime64|M8)\[(?P<unit>.+), (?P<tz>.+)\]")
     _cache = {}
+    # TODO: restore caching? who cares though? It seems needlessly complex.
+    # np.dtype('datetime64[ns]') isn't a singleton
```

Comment:
this is a huge performance penalty w/o caching

Reply:
Did you see the perf numbers I posted in 
https://github.com/pandas-dev/pandas/pull/23990#issuecomment-442811414? It seems to be slightly faster without caching (though within noise).

Reply:
try running a good part of the test suite. Its the repeated consruction that's a problem, not the single contruction which is fine. W/o caching you end up creating a huge number of these